avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -954,6 +954,10 @@ private void addAvroRecordField(
       defaultVal = fieldSchema.defaultValue();
       if (defaultVal instanceof Byte) {
         defaultVal = ((Byte) defaultVal).intValue();
+      } else if (defaultVal instanceof Short) {
+        defaultVal = ((Short) defaultVal).intValue();
+      } else if (defaultVal instanceof ByteBuffer) {
+        defaultVal = ((ByteBuffer) defaultVal).array();
```

Comment:
just mentioning here that `array` is optional. Are we sure it always has it? (not a direct/read-only bytebuffer?)
Do we need to call `hasArray` for the sake of completeness or we can omit? https://docs.oracle.com/javase/7/docs/api/java/nio/ByteBuffer.html#array()

Maybe a comment in this case might be good.

Reply:
Great catch, @kkonstantine. Removed use of `array()` altogether and using `duplicate().get(...)` instead since the latter will work without side effects on all ByteBuffer implementations. 

(I decided against using `hasArray()` and `array()`, since that should be used with caution since it doesn't always work. And, if we did that, we'd have two code paths that would have to be tested. The current approach is simpler and easier to test.)